public/docs/ts/latest/cookbook/validation.jade

```diff
@@ -0,0 +1,253 @@
+<a id="template-driven"></a>
+:marked
+   ## Template-Driven
+
+   Using the template-driven approach to form validation, the validation is defined in the template.
+   Each control on the form defines its validation, binding, and validation messages in the template.
+
++makeExample('cb-validation/ts/app/hero-form-template.component.html','name-with-error-msg','app/hero-form-template.component.html')
+
+:marked
+   Here we define a standard label and set up an input box for validation as follows:
+      - Add the desired HTML validation attributes. In this example, 
+      we add `required`, `minlength`, and `maxlength` attributes on the input box to define the validation rules. 
+      - Use `ngModel` binding to set the default value and track the user's changes to the input box.
+      - Set `ngControl` to register the input box as a control with the `ngForm` directive.
+      - Define a local variable (`name` in this example) that references the control registered with the `ngForm` directive.
+
+   We use the local variable to determine whether to display a validation message.
+      - We use `*ngIf` to check whether the control has errors and whether it is `dirty` or `touched` before
+      displaying the validation block `div`.
+      Adding the check for `dirty` or `touched` prevents display of errors before the user has a chance to edit the
+      value. This is most useful when adding new data, such as a new hero.
+      - We then have a separate `div` for each possible validation error. In our example, we defined validation for
+      `required`, `minlength`, and `maxlength`, so we add one `div` for each one. Each `div` is marked with `[hidden]`
+      so the validation message is hidden unless the control has the specified error.
+
+   Repeat for each data entry control on the form.
+
+   The component class then manages the model used in the data binding. And provides any other code required
+   by the view.
+
++makeExample('cb-validation/ts/app/hero-form-template.component.ts','class','app/hero-form-template.component.ts')
+
+:marked
+   Use this template-driven validation technique when working with simple forms with simple validation scenarios.
+
+   Here's the complete solution for the template-driven approach:
+
++makeTabs( 
+  `cb-validation/ts/app/main.ts,
+   cb-validation/ts/app/app.component.ts,
+   cb-validation/ts/app/hero.ts,
+   cb-validation/ts/app/hero-form-template.component.html,
+   cb-validation/ts/app/hero-form-template.component.ts`, 
+  '', 
+  'app/main.ts, app/app.component.ts, app/hero.ts, app/hero-form-template.component.html, app/hero-form-template.component.ts' )
+
+.l-main-section
+<a id="model-driven-binding"></a>
+:marked
+   ## Model-Driven with Data Binding
+
+   Using the model-driven approach to form validation, the validation rules are specified in the model as defined in the
+   component class. Defining the validation in the class instead of the template gives you more control. You
```

Comment:
Does this technique produce rendered HTML equivalent to the template-driven solution? If not, I think that needs to be addressed in some way - either by updating the code so both techniques produce the same HTML or mentioning it in here somewhere.